src/model/Dough.java

```diff
@@ -1,9 +1,69 @@
 package model;
 
+import java.util.ArrayList;
+
 public class Dough extends Ingredient{
+    private ArrayList<Ingredient> ingredients;
+    private int freeSize = this.getSize();
     private enum DoughType {
         NEW_YORK,
         NEAPOLITAN,
         SICILIAN
     }
+
+    public Dough(int price, String name, int size, ArrayList<Ingredient> ingredients) {
+        super(price, name, size);
+        this.ingredients = ingredients;
+        this.setPrice(price + this.findSumPrice());
+    }
+
+    public boolean addIngredient(Ingredient ingredient) {
+        if (ingredient == null || ingredient.getSize() > this.getFreeSize()) {
+            return false;
+        }
+
+        ingredients.add(ingredient);
+        this.setFreeSize(this.getFreeSize() - ingredient.getSize());
+        this.setPrice(this.getPrice() + ingredient.getPrice());
+        return true;
+    }
+
+    public boolean removeIngredient(Ingredient ingredient) {
+        if (ingredient == null || !this.ingredients.contains(ingredient)) {
+            return false;
+        }
+
+        ingredients.remove(ingredient);
+        this.setFreeSize(this.getFreeSize() + ingredient.getSize());
+        this.setPrice(this.getPrice() - ingredient.getPrice());
```

Comment:
according to the task price is a pure price of dough
to get total price we use findSumPrice
do not change price here